### PR TITLE
Evaluate js in call frame

### PR DESCRIPTION
**Pull request by alan-knight**

### CODE REVIEW

File: dwds/lib/src/utilities/objects.dart [hunk 1]
```diff
@@ -10,18 +10,29 @@ import 'package:webkit_inspection_protocol/webkit_inspection_protocol.dart';
...
-  Map<String, dynamic> get rawValue => _map['value'] as Map<String, dynamic>;
+  Object get rawValue => _map == null ? null : _map['value'];
 
   /// Remote object value in case of primitive values or JSON values (if it was
   /// requested). (optional)
-  RemoteObject get value => RemoteObject(rawValue);
+  RemoteObject get value {
+    if (_remoteObjectValue != null) return _remoteObjectValue;
+    if (rawValue == null) return null;
+    var value = _map['value'];
```
Comment (jakemac53) [inline dwds/lib/src/utilities/objects.dart]: nit: don't shadow the instance getter here, `val` would probably be fine?

File: dwds/test/variable_scope_test.dart [hunk 2]
```diff
@@ -56,13 +56,20 @@ void main() {
...
       await service.removeBreakpoint(isolateId, bp.id);
       var stack = await service.getStack(isolateId);
-      // Resume as to not impact other tests.
-      await service.resume(isolateId);
       return stack;
     }
 
+    Stack stack;
+
+    setUp(() async {
```
Comment (jakemac53) [inline dwds/test/variable_scope_test.dart]: There is already a `setUp` method here, apparently you can have multiple so thats news to me but we should combine  these :)

Comment (alan-knight) [reply dwds/test/variable_scope_test.dart]: Done.

File: dwds/lib/src/utilities/objects.dart (hunk 1 above)
Comment (alan-knight) [reply dwds/lib/src/utilities/objects.dart]: Done.

**Pull request merged**